pubsub_logging/utils.py

```diff
@@ -49,23 +49,39 @@ def compat_urlsafe_b64encode(v):
         return base64.urlsafe_b64encode(v)
 
 
-def get_pubsub_client(http=None):  # pragma: NO COVER
-    """Return a thread local Pub/Sub client.
+def get_pubsub_client(http=None, credentials=None):
+    """Return a Pub/Sub client.
 
     Args:
       http: httplib2.Http instance. Defaults to None.
     Returns:
       Cloud Pub/Sub client.
     """
-    if not hasattr(clients, 'client'):
+    if not credentials:
         credentials = GoogleCredentials.get_application_default()
-        if credentials.create_scoped_required():
-            credentials = credentials.create_scoped(PUBSUB_SCOPES)
-        if not http:
-            http = httplib2.Http()
-        credentials.authorize(http=http)
-        clients.client = discovery.build('pubsub', 'v1beta2', http=http)
-    return clients.client
+    if credentials.create_scoped_required():
+        credentials = credentials.create_scoped(PUBSUB_SCOPES)
+    if not http:
+        http = httplib2.Http()
+    credentials.authorize(http=http)
+    return discovery.build('pubsub', 'v1beta2', http=http)
+
+
+def check_topic(client, topic, retry):
```

Comment:
Er, maybe this should return True/False if the topic exists rather than just raising an HttpError? That seems kind of weird.